integration/yarn/src/main/java/alluxio/yarn/ApplicationMaster.java

```diff
@@ -397,6 +410,39 @@ private void launchWorkerContainer(Container container) {
     }
   }
 
+  /**
+   * Checks if an Alluxio master node is already running
+   * or not on the master address given.
+   *
+   * @return true if master exists, false otherwise
+   */
+  private boolean masterExists() {
+
+    String webPort = Configuration.get(PropertyKey.MASTER_WEB_PORT);
+
+    try {
+      URL myURL = new URL("http://" + mMasterAddress + ":" + webPort + Constants.REST_API_PREFIX + "/master/version");
+      LOG.debug("Checking for master at: " + myURL.toString());
+      HttpURLConnection connection = (HttpURLConnection) myURL.openConnection();
+      connection.setRequestMethod(HttpMethod.GET);
+      int resCode = connection.getResponseCode();
+      LOG.debug("Response code from master was: " + Integer.toString(resCode));
+
+      if (resCode == HttpURLConnection.HTTP_OK) {
+        connection.disconnect();
+        return true;
+      } else {
+        connection.disconnect();
```

Comment:
We can further improve this to:
```
return resCode == HttpURLConnection.HTTP_OK
```
And remove the return statement at this end of this function.